File: CS/NetCore/net/06_Display/Ellipsis.cs
namespace InputManWin13_Demo._06_Display;

public partial class Ellipsis : InputManWin13_Demo.DemoBase_settings
{
    public Ellipsis()
    {
        InitializeComponent();
        InitializeForm();
    }

    private void InitializeForm()
    {
        // ***** 初期設定

        // Ellipsisプロパティ
        gcComboBox1.SelectedIndexChanged += (s, e) => SetEllipsis();

        // EllipsisStringプロパティ
        gcTextBox2.TextChanged += (s, e) => SetEllipsisString();

        // ヘッダの省略文字
        checkBox1.CheckedChanged += (s, e) => SetListColumnsEllipsis();

        // リストの省略文字
        checkBox2.CheckedChanged += (s, e) => SetListItemsEllipsis();

        // ***** 初期値
        gcComboBox1.SelectedIndex = Convert.ToInt32(gcTextBox1.Ellipsis);
        gcTextBox2.Text = gcTextBox1.EllipsisString;
        checkBox1.Checked = gcComboBox2.ListColumns[0].Header.Ellipsis;

        gcComboBox2.ShowOverflowTip = false;
        gcComboBox2.DropDown.Width = gcComboBox2.Width;
        gcComboBox2.ListColumns[0].Width = gcComboBox2.Width;
    }

    private void SetEllipsis()
    {
        gcTextBox1.Ellipsis = (GrapeCity.Win.Editors.EllipsisMode)gcComboBox1.SelectedIndex;
        gcTextBox3.Ellipsis = (GrapeCity.Win.Editors.EllipsisMode)gcComboBox1.SelectedIndex;
        gcComboBox2.Ellipsis = (GrapeCity.Win.Editors.EllipsisMode)gcComboBox1.SelectedIndex;
    }

    private void SetEllipsisString()
    {

        gcTextBox1.EllipsisString = gcTextBox2.Text;
        gcTextBox3.EllipsisString = gcTextBox2.Text;
        gcComboBox2.EllipsisString = gcTextBox2.Text;
    }

    private void SetListColumnsEllipsis()
    {
        gcComboBox2.ListColumns[0].Header.Ellipsis = checkBox1.Checked;
    }

    private void SetListItemsEllipsis()
    {
        for (int i = 0; i < gcComboBox2.Items.Count; i++)
        {
            gcComboBox2.Items[i].SubItems[0].Ellipsis = checkBox2.Checked;
        }
    }
}

File: VB/NetCore/net/06_Display/Ellipsis.vb
Namespace _06_Display
    Partial Public Class Ellipsis
        Inherits InputManWin13_Demo.DemoBase_settings
        Public Sub New()
            InitializeComponent()
            InitializeForm()
        End Sub

        Private Sub InitializeForm()
           ' ***** 初期値
            gcComboBox1.SelectedIndex = Convert.ToInt32(gcTextBox1.Ellipsis)
            gcTextBox2.Text = gcTextBox1.EllipsisString
            checkBox1.Checked = gcComboBox2.ListColumns(0).Header.Ellipsis

            gcComboBox2.ShowOverflowTip = False
            gcComboBox2.DropDown.Width = gcComboBox2.Width
            gcComboBox2.ListColumns(0).Width = gcComboBox2.Width
        End Sub

        Private Sub gcComboBox1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles gcComboBox1.SelectedIndexChanged
            ' Ellipsisプロパティ
            gcTextBox1.Ellipsis = CType(gcComboBox1.SelectedIndex, GrapeCity.Win.Editors.EllipsisMode)
            gcTextBox3.Ellipsis = CType(gcComboBox1.SelectedIndex, GrapeCity.Win.Editors.EllipsisMode)
            gcComboBox2.Ellipsis = CType(gcComboBox1.SelectedIndex, GrapeCity.Win.Editors.EllipsisMode)
        End Sub

        Private Sub gcTextBox2_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles gcTextBox2.TextChanged
            ' EllipsisStringプロパティ
            gcTextBox1.EllipsisString = gcTextBox2.Text
            gcTextBox3.EllipsisString = gcTextBox2.Text
            gcComboBox2.EllipsisString = gcTextBox2.Text
        End Sub

        Private Sub checkBox1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkBox1.CheckedChanged
            ' ヘッダの省略文字
            gcComboBox2.ListColumns(0).Header.Ellipsis = checkBox1.Checked
        End Sub

        Private Sub checkBox2_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkBox2.CheckedChanged
            ' ヘッダの省略文字
            For i As Integer = 0 To gcComboBox2.Items.Count - 1
                gcComboBox2.Items(i).SubItems(0).Ellipsis = checkBox2.Checked
            Next

        End Sub
    End Class
End Namespace
